core/drivers/atmel_wdt.c

```diff
@@ -240,33 +242,50 @@ static TEE_Result wdt_node_probe(const void *fdt, int node,
 	if (!wdt)
 		return TEE_ERROR_OUT_OF_MEMORY;
 
+	it_hdlr = calloc(1, sizeof(*it_hdlr));
+	if (!it_hdlr) {
+		free(wdt);
```

Comment:
Please use the common cleanup path instead.

Reply:
First, we must be 100% sure that pointer aliasing never will become a problem. I'm not sure that's the case at the moment.

Reply:
I get your point. There would need some casting.
Forget it.